invenio/modules/workflows/client.py

```diff
@@ -38,79 +31,24 @@ def run_workflow(wfe, data, stop_on_halt=False,
     :param kwargs: additionnal keywords arguements
     :param wfe: workflow engine in charge of workflow execution
     """
-    while True:
-        try:
-            if initial_run:
-                initial_run = False
-                wfe.process(data)
-                break
-            else:
-                wfe.restart('next', 'first')
-                break
-        except WorkflowHalt as workflowhalt_triggered:
-            current_obj = wfe.get_current_object()
-            if current_obj:
-                if workflowhalt_triggered.action:
-                    current_obj.set_action(workflowhalt_triggered.action,
-                                           workflowhalt_triggered.message)
-                    current_obj.version = ObjectVersion.HALTED
-                    current_obj.save(version=ObjectVersion.HALTED,
-                                     task_counter=wfe.getCurrTaskId(),
-                                     id_workflow=wfe.uuid)
-                else:
-                    current_obj.version = ObjectVersion.WAITING
-                    current_obj.save(version=ObjectVersion.WAITING,
-                                     task_counter=wfe.getCurrTaskId(),
-                                     id_workflow=wfe.uuid)
-            else:
-                wfe.log.warning("No active object found!")
-
-            wfe.save(status=WorkflowStatus.HALTED)
-            message = "Workflow '%s' halted at task %s with message: %s" % \
-                      (wfe.name,
-                       wfe.get_current_taskname() or "Unknown",
-                       workflowhalt_triggered.message)
-            wfe.log.warning(message)
-            if stop_on_halt:
-                break
-        except Exception as exception_triggered:
-            msg = "Error: %r\n%s" % \
-                  (exception_triggered, traceback.format_exc())
-            wfe.log.error(msg)
-            current_obj = wfe.get_current_object()
-            if current_obj:
-                # Sets an error message as a tuple (title, details)
-                current_obj.set_error_message((str(exception_triggered), msg))
-                current_obj.save(
-                    ObjectVersion.ERROR,
-                    wfe.getCurrTaskId(),
-                    id_workflow=wfe.uuid
-                )
-            wfe.save(status=WorkflowStatus.ERROR)
-            if stop_on_error:
-                if isinstance(exception_triggered, WorkflowError):
-                    raise exception_triggered
-                else:
-                    raise WorkflowError(
-                        message=msg,
-                        id_workflow=wfe.uuid,
-                        id_object=wfe.getCurrObjId(),
-                    )
+    # FIXME: module_name does nothing
```

Comment:
If tests are the only use-case for "updating `extra_data` while calling eng.process", then I will work around this purely in the tests' code and drop this `FIXME`.

PS I have grown very wary of `**kwargs` and `*args` (bitten too many times to count!). All arguments should have a name that declares its purpose.